R/process_lookup_homelessness.R

```diff
@@ -0,0 +1,114 @@
+# It would probably be simplest to create a lookup from the Homelessness data of one row per CHI +
+#   start dates of all applications. This can then be matched to episode and individual to easily create the hl1_in_fy flag(s)
+# (if someone is in the lookup -> 1, otherwise -> 0). For the episode file, it is slightly more complicated to create the other flags,
+# either matching on all the data by CHI and then doing some comparison of dates (as in SPSS),
+# or using some of the fancy new joins (dplyr 1.1.0 joins) and doing it that way.
+#
+#
+
+
+# data <- slfhelper::read_slf_episode("1718", c("anon_chi", "record_keydate1", "record_keydate2", "recid"))
+
+# year <- "1718"
+
+
+#' create homelessness lookup
+#' @description reads in homelessness extract and selects CHIs and flags them as homeless in FY
+#'
+#' @param year The year to process, in FY format.
+#' @return the final data as a [tibble][tibble::tibble-package].
+#' @export
+#' @family process extracts
+#'
+create_homelessness_lookup <- function(year) {
+  homelessness_lookup <- read_file(get_source_extract_path(year, "Homelessness")) %>%
```

Comment:
Doing it this way means it will work the same if you just call it `create_homelessness_lookup` as it will use `read_file(get_source_extract_path(year, "Homelessness"))` by default for `homelessness_data`. However in the targets pipeline we can pass the homelessness data directly, this will create a dependency, so targets knows the homelessness data has to be processed first before it can do this lookup.
```suggestion
create_homelessness_lookup <- function(
year, 
homelessness_data = read_file(get_source_extract_path(year, "Homelessness"))
) {
  homelessness_lookup <- homelessness_data %>%
```